### PR TITLE
chore: Fix linter findings for `revive:enforce-slice-style` in `plugins/inputs/[k-o]*`

### PR DESCRIPTION
**Pull request by zak-pawel**
## Summary

Address findings for [revive:enforce-slice-style](https://github.com/mgechev/revive/blob/master/RULES_DESCRIPTIONS.md#enforce-slice-style) in `plugins/inputs/[k-o]*`.

As we agreed here https://github.com/influxdata/telegraf/issues/15812, we will fix the errors found by the following configuration:
```
      - name: enforce-slice-style
        arguments: ["make"]
```

It is only part of the bigger work.
After all findings of this type in whole project are handled, we can enable `revive:enforce-slice-style` rule in `golangci-lint`.

## Checklist

- [x] No AI generated code was used in this PR

### CODE REVIEW

**Review by DStrand1 — APPROVED** [review]

File: plugins/inputs/mongodb/mongodb_data.go [hunk 1]
```diff
@@ -34,7 +34,7 @@ func NewMongodbData(statLine *statLine, tags map[string]string) *MongodbData {
 		StatLine: statLine,
 		Tags:     tags,
 		Fields:   make(map[string]interface{}),
-		DbData:   []DbData{},
+		DbData:   make([]DbData, 0),
```
Comment (srebhan) [inline plugins/inputs/mongodb/mongodb_data.go:37]: Remove?

**Review by srebhan — COMMENTED** [review]
Thanks a lot @zak-pawel! A few potential removal in the comments, please let me know if they are necessary!

File: plugins/inputs/kubernetes/kubernetes.go [hunk 2]
```diff
@@ -51,7 +51,7 @@ type Kubernetes struct {
 func init() {
 	inputs.Add("kubernetes", func() telegraf.Input {
 		return &Kubernetes{
-			LabelInclude: []string{},
+			LabelInclude: make([]string, 0),
```
[the remark this answers is not in the document]
Comment (zak-pawel) [reply plugins/inputs/kubernetes/kubernetes.go]: Hope it isn't :)

File: plugins/inputs/modbus/configuration_request.go [hunk 3]
```diff
@@ -230,10 +230,10 @@ func (c *ConfigurationPerRequest) Process() (map[byte]requestSet, error) {
...
 		if !found {
 			set = requestSet{
-				coil:     []request{},
-				discrete: []request{},
-				holding:  []request{},
-				input:    []request{},
+				coil:     make([]request, 0),
+				discrete: make([]request, 0),
+				holding:  make([]request, 0),
+				input:    make([]request, 0),
```
[the remark this answers is not in the document]
Comment (zak-pawel) [reply plugins/inputs/modbus/configuration_request.go]: Probably

File: plugins/inputs/modbus/request.go [hunk 4]
```diff
@@ -28,7 +28,7 @@ func splitMaxBatchSize(g request, maxBatchSize uint16) []request {
 	idx := 0
 	for start := g.address; start < g.address+g.length; {
 		current := request{
-			fields:  []field{},
+			fields:  make([]field, 0),
```
[the remark this answers is not in the document]
Comment (zak-pawel) [reply plugins/inputs/modbus/request.go]: Removed

File: plugins/inputs/modbus/request.go [hunk 5]
```diff
@@ -238,7 +238,7 @@ func groupFieldsToRequests(fields []field, params groupingParams) []request {
 			groups = append(groups, current)
 		}
 		current = request{
-			fields:  []field{},
+			fields:  make([]field, 0),
```
[the remark this answers is not in the document]
Comment (zak-pawel) [reply plugins/inputs/modbus/request.go]: Removed

File: plugins/inputs/mongodb/mongodb_data.go (hunk 1 above)
Comment (zak-pawel) [reply plugins/inputs/mongodb/mongodb_data.go]: Removed

File: plugins/inputs/neptune_apex/neptune_apex_test.go [hunk 6]
```diff
@@ -35,7 +35,7 @@ func TestGather(t *testing.T) {
 		},
 		{
 			name:    "Good case, 0 servers",
-			servers: []string{},
+			servers: make([]string, 0),
```
[the remark this answers is not in the document]
Comment (zak-pawel) [reply plugins/inputs/neptune_apex/neptune_apex_test.go]: Removed

**Review by srebhan — APPROVED** [review]
Thanks @zak-pawel!

**Pull request merged**